src/unix/lwt_unix_unix.h

```diff
@@ -1860,7 +1851,7 @@ static value result_readdir_n(struct job_readdir_n *job)
         result = caml_alloc(job->count, 0);
         long i;
         for (i = 0; i < job->count; i++) {
-            Store_field(result, i, caml_copy_string(job->entries[i].d_name));
+            Store_field(result, i, caml_copy_string(job->entries[i]));
```

Comment:
That's the thing I had doubts about. But then, the previous code did not try to deallocate any of the `dirent`s. Was there a memory leak before? What's the difference in the code that makes that change necessary?